READMEにテーブル情報を記述

ryousuke-515 opened the pull request:
# What
READMEにテーブル情報を記述

# Why
データベース設計を行うため

# ER図のURL
https://gyazo.com/58dccee3e02e347ad27a322c081406db

ryousuke-515 committed:
READMEにテーブル情報を記述

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,47 @@
+# テーブル設計
 
-Things you may want to cover:
+## users テーブル
```
必要なカラムが足りていないので、見本アプリを見直して追加しましょう！

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,47 @@
+| password | string | null: false |
 
-* System dependencies
+## items テーブル
```
必要なカラムが足りていないので、見本アプリを見直して追加しましょう！

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,47 @@
+- belongs_to :item
+- has_one :address
+
+## addresses テーブル
```
必要なカラムが足りていないので、見本アプリを見直して追加しましょう！

syuji-yamamoto reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認ください！

ryousuke-515 committed:
READMEにテーブル情報を記述（追加）

ryousuke-515 committed:
Revert "READMEにテーブル情報を記述（追加）"

This reverts commit f4cea450019f3134d6de1cafd1041af87a8b3859.

miyagit commented on README.md:
```diff
@@ -1,24 +1,47 @@
+### Association
 
-* Database initialization
+- has_one :buys
```
has_oneのあとは単数系にしましょう。

miyagit commented on README.md:
```diff
@@ -1,24 +1,47 @@
+- has_one :buys
 
-* How to run the test suite
+## buys テーブル
```
こちらbuysテーブルではなく、item_purchasesテーブルの方が意図がわかりやすいので、item_purchasesテーブルに修正しましょう。

miyagit reviewed (commented):
レビュー依頼ありがとうございます。
こちらカラム情報がまだ不足している状態なので、もう一度見本アプリケーションを見直していただけるとありがたいです。

ryousuke-515 committed:
READMEにテーブル情報を記述（追加2）

miyagit commented on README.md:
```diff
@@ -1,24 +1,63 @@
+| Column              | Type            | Options                        |
+| ------              | ------          | -----------                    |
+| user                | references      | null: false, foreign_key: true |
+| images              | integer         | null: false                    |
```
今回のアプリケーションでは、ActiveStorageを使うのでimageカラムは必要なくなります。
なので、imageカラムは削除しましょう。ActiveStorageについては下記に公式サイトを載せたので、良ければ参考にして頂ければと思います。
https://railsguides.jp/active_storage_overview.html

miyagit commented on README.md:
```diff
@@ -1,24 +1,63 @@
+| user                | references      | null: false, foreign_key: true |
+| images              | integer         | null: false                    |
+| description         | text            | null: false                    |
+| category            | integer         | null: false                    |
+| condition           | integer         | null: false                    |
+| shipping_charges    | integer         | null: false                    |
+| area                | integer         | null: false                    |
+| shipping_days       | integer         | null: false                    |
```
こちらactive_hashで実装するところだと思うので、カラム名_idの形にしましょう。
よければ下記のカリキュラムを参考にして頂ければと思います。
https://master.tech-camp.in/curriculums/4999

miyagit commented on README.md:
```diff
@@ -1,24 +1,63 @@
 
-* Database initialization
+### Association
 
```
下記のサイトを参考にactive_hashのAssociationを追加しましょう。
https://master.tech-camp.in/curriculums/4999

ryousuke-515 replied:
iamgeカラムの削除。
active_hashを実装するカラム名に_idを追記する。
active_hashのアソシエーションを記述する。
ER図になります↓
https://gyazo.com/5d32214398f92649a7e052af0a6837d3

ryousuke-515 committed:
READMEにテーブル情報を記述（追加３）

active_hashで実装するカラム名に_idを追記。
active_hashのアソシエーションを記述。

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,62 @@
+# テーブル設計
 
-Things you may want to cover:
+## users テーブル
```
他のテーブルとのアソシエーションを記述しましょう！

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,62 @@
+| birth_date      | date   | null: false |
 
-* System dependencies
+## items テーブル
```
商品名と商品説明を保存するカラムを用意しましょう！

ryousuke-515 committed:
Update README.md

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,62 @@
+| ------          | ----------       | ------------------------------ |
+| user            | references       | null: false, foreign_key: true |
+| item            | references       | null: false, foreign_key: true |
+| address         | references       | null: false, foreign_key: true |
```
商品の購入管理はuserとitemのidのみで管理が可能です。
したがって、それ以外のカラムは今回は不要です！

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,62 @@
+| post_code         | string           | null: false                    |
+| item_purchases_id | integer          | null: false, foreign_key: true |
+| city              | string           | null: false                    |
+| building_name     | string           | null: false                    |
```
番地を保存するカラムが足りていないので用意してあげましょう！

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,62 @@
+### Association
+
+- belongs_to :item
+- has_one :address
```
usersテーブルとのアソシエーションを追加しましょう！

syuji-yamamoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点を確認させていただきました！
コメントしましたので、ご確認ください！

ryousuke-515 committed:
READMEにテーブル情報を記述（追加4）

必要なカラムの追加と、不要なカラムの削除。

syuji-yamamoto commented on README.md:
```diff
@@ -15,11 +15,17 @@
 | Column              | Type            | Options                        |
 | ------              | ------          | -----------                    |
 | user                | references      | null: false, foreign_key: true |
+| item_name           | text            | null: false
```
テーブル名で判断できるので、カラム名のitem_は不要です！
また情報量を考慮すると、string型で十分です！

syuji-yamamoto commented on README.md:
```diff
@@ -54,6 +59,7 @@
 | item_purchases_id | integer          | null: false, foreign_key: true |
 | city              | string           | null: false                    |
+| address           | string           | null: false                    |
 | building_name     | string           | null: false                    |
```
今回の実装では、建物番号の入力は任意になるため、not null制約はなくても大丈夫です！

syuji-yamamoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点を確認させていただきました！
コメントしましたので、ご確認ください！

```
商品説明のカラムはdescriptionですでに指定していますが、名前を変えた方がよいでしょうか？
```
大変失礼致しました！見落としておりました。
こちらこのままで大丈夫です！

ryousuke-515 committed:
READMEにテーブル情報を記述（追加5）

itemテーブルのitem_nameをnameに変更。
ビルの名前は任意であるため、not null制約を解除。

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,68 @@
+- belongs_to :item
+- has_one :address
+
+## addresses テーブル
```
都道府県を保存するカラムをinteger型で用意しましょう！active_hashを使用して実装します！

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,68 @@
+
+### Association
+
+- belongs_to :buy
```
item_purchasesテーブルとのアソシエーションに変えておきましょう！

syuji-yamamoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点を確認させていただきました！
追加でコメントしましたので、ご確認をお願いします！

ryousuke-515 committed:
READMEにテーブル情報記述（追加6）

addressesテーブルに都道府県の入力を保存するカラムを追加。
addressesテーブルのアソシエーションがbuyのままだったので、item_purchasenに変更。

syuji-yamamoto commented:
レビュー依頼ありがとうございます！
修正点を確認させていただきました！
LGTMとさせていただきます！実装お疲れ様でした！
引き続き、実装頑張っていきましょう！
